Right Stick controls IR pointer

goedzo opened the pull request:
This allows the Wii/IR pointer to be controlled by the right analog stick.  Fixes #79

goedzo committed:
Right Stick controls IR pointer

This allows the Wii/IR pointer to be controlled by the right analog stick.  Fixes #79

flyinghead commented:
With this change, the IR pointer cannot be controlled by a mouse/pointer anymore. This is not going to work for light gun games and such.
A core option to select between mouse or right analog stick would be better. One of the Nintendo DS cores has such an option.

goedzo commented:
Interesting suggestion however i am new to this core for development. Could you give me some pointers on where in the code i can add an extra core option? I will then try to adapt the code to use it.

flyinghead commented:
You need to define a new core option in ```Source/Core/DolphinLibretro/Options.h/cpp```.
Then use this option's value in ```Source/Core/DolphinLibretro/Input.cpp```to bind the IR pointer to the mouse or right analog stick.

goedzo committed:
Configure a core option to determine IR behaviour

This allows a core option to allow right stick to control IR pointer, or use the mouse (light-gun) instead.

goedzo commented:
Thanks! Implemented this now as a core option. Please merge now :).

flyinghead commented on Source/Core/DolphinLibretro/Options.cpp:
```diff
@@ -148,7 +148,8 @@ bool Option<T>::Updated()
-Option<LogTypes::LOG_LEVELS> logLevel("dolphin_log_level", "Log Level",
+Option<int> irMode("dolphin_ir_mode", "IR Mode", 1,
+                     {"Right Stick controls pointer", "Mouse controls pointer"});Option<LogTypes::LOG_LEVELS> logLevel("dolphin_log_level", "Log Level",
                                       {{"Info", LogTypes::LINFO},
```
Missing `\n`

goedzo replied:
Fixed

flyinghead commented on Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp:
```diff
@@ -30,13 +32,25 @@ Cursor::Cursor(const std::string& name_) : ControlGroup(name_, GroupType::Cursor
+    numeric_settings.emplace_back(std::make_unique<NumericSetting>(_trans("Width"), 0.5));
+    numeric_settings.emplace_back(std::make_unique<NumericSetting>(_trans("Height"), 0.5));
+    numeric_settings.emplace_back(std::make_unique<NumericSetting>(_trans("Dead Zone"), 0, 0, 20));
+    boolean_settings.emplace_back(std::make_unique<BooleanSetting>(_trans("Relative Input"), true));
```
Relative input was false for screen pointer/mouse.
Auto-hide was false.
Any reason to change these values?

goedzo replied:
Nope, just  copy and paste error here.

goedzo committed:
Update Cursor.cpp

goedzo committed:
Update Options.cpp

goedzo commented:
Fixed all comments. Thanks for the review!

flyinghead reviewed (approved):
Looks good to me
